### PR TITLE
Add typehints to more modules

### PR DESCRIPTION
**Pull request by njgheorghita**
### What was wrong?
More modules need type-hints

### How was it fixed?
Added type-hints to a couple more modules

### CODE REVIEW

File: eth_utils/hexadecimal.py [hunk 1]
```diff
@@ -4,46 +4,48 @@
...
     is_string,
     is_text,
 )
 
 
-def decode_hex(value):
+def decode_hex(value: str) -> bytes:
     if not is_text(value):
         raise TypeError('Value must be an instance of str')
     return codecs.decode(remove_0x_prefix(value), 'hex')
```
Comment (njgheorghita) [inline eth_utils/hexadecimal.py]: Not sure what the best move is here? According to https://github.com/python/typeshed/issues/300 - it seems like mypy won't support accepting strings to `decode()` regardless of the encoding argument. . .

File: eth_utils/hexadecimal.py [hunk 2]
```diff
@@ -4,46 +4,48 @@
...
 
 
-def encode_hex(value):
+def encode_hex(value: str) -> str:
     if not is_string(value):
         raise TypeError('Value must be an instance of str or unicode')
     binary_hex = codecs.encode(value, 'hex')
```
Comment (njgheorghita) [inline eth_utils/hexadecimal.py]: Same issue here

File: eth_utils/hexadecimal.py (hunk 1 above)
Comment (pipermerriam) [reply eth_utils/hexadecimal.py]: It's fine to add a `# type: ignore` to this with a comment linking to the offending issue on github.

File: eth_utils/hexadecimal.py (hunk 2 above)
Comment (pipermerriam) [reply eth_utils/hexadecimal.py]: Same, use `# type: ignore` here.

File: eth_utils/address.py [hunk 3]
```diff
@@ -119,7 +119,7 @@ def is_same_address(left: AnyAddress, right: AnyAddress) -> bool:
         return to_normalized_address(left) == to_normalized_address(right)
 
 
-def to_checksum_address(value: AnyStr) -> Address:
+def to_checksum_address(value: AnyStr) -> HexAddress:
```
Comment (pipermerriam) [inline eth_utils/address.py]: Hrm, this actually looks like it should be `ChecksumAddress` as the return type.

Comment (njgheorghita) [reply eth_utils/address.py]: Ahh, yup!! Made the change

**Comment by njgheorghita** [comment]
@pipermerriam changed the custom types back to using `NewType` - should be good to merge

**Pull request merged**